### ISSUE TITLE
[feature] shortcuts

### ISSUE THREAD

**Issue by jrpie** [jrpie/launcher#45]
Support for [shortcuts](https://developer.android.com/develop/ui/views/launch/shortcuts) 
would be nice. These should be displayed in a separate list.
[Pinned shortcuts](https://developer.android.com/develop/ui/views/launch/shortcuts/creating-shortcuts#pinned) should be shown in the same way as apps.

It should be possible to bind shortcuts to gestures.

There does not seem to be a good documentation for this. Implementation in Launcher3:
https://android.googlesource.com/platform/packages/apps/Launcher3/+/8b2002e28072c52d78f6d052c6ff6da50a2d0324
https://android.googlesource.com/platform/packages/apps/Launcher3/+/278359539c02ca160caf1df54ce96053a2a2ef59

**Comment by wassupluke** [comment]
Would this provide functionality to see icons/shortcuts made by using a browsers "Add to Home Screen" feature, or would that be a separate feature request?  Asking because I like to keep icons/buttons to click for easy access to my most used sites (e.g., jumping straight to mypihole's admin page).

**Comment by jrpie** [comment]
Yes, that is what this feature is supposed to solve. (Shortcuts like those created by "add to home screen" are called pinned shortcuts by Android)

**Comment by jrpie** [comment]
I have not decided on the UI for this yet:
1. Treat pinned shortcuts just like apps. Adding them just adds them to the list of apps. The user can then bind this shortcut to a gesture in the same way as an app. This might be tedious, since in order to bind a shortcut only to a gesture, a user would have to:
* add the shortcut
* go to settings, pick an "app" for the gesture and search for the newly added shortcut in the list 
* hide the shortcut
2. When a user wants to add a shortcut, ask whether to (a) add it to the app list or (b) bind it to a specific gesture. Here the initial setup it a lot cleaner, but it is not possible to move shortcuts to a different gesture later.

**Comment by BANOnotIT** [comment]
Can't you add it to the app automatically but show a modal/view asking to chose gesture?

**Comment by jrpie** [comment]
That sound like a good solution 👍🏼 I could also add an entry "bind to gesture" to the context menu that opens the same dialog.

**Comment by GSI** [comment]
essential feature!

**Comment by GSI** [comment]
Can the quick profile switcher be reached directly as an action, without having to pull down quick settings and having to hit the icon of the current profile (the usual path)?

**Comment by jrpie** [comment]
Nice idea! I don't know whether this is possible. Opened a separate issue (#99) as this is off topic here.

**Comment by jrpie** [comment]
Basic support for this featurewas added; shortcuts can now be bound to gestures. Support for showing shortcuts in the app list is planned but not yet implemented, as it requires some changes to the storage format.

**Comment by GSI** [comment]
Excellent!

.. where can one donate some Monero?

**Comment by jrpie** [comment]
> .. where can one donate some Monero?

Thank you for considering to donate, really appreciate it! I still need to make up my mind on how to handle donations (probably I should put something in the README). On the one hand, not accepting donations would feel ungrateful, on the other hand I feel like other open source projects (or charities) need money more urgently. My financial situation is quite good (there are no tuition fees in Germany!). Also this is more of a hobby project, and I feel like taking money for it might spoil the fun.

Regarding cryptocurrencies, even though I like the concept from a theoretical perspective, I'm reluctant to participate in any system built on proof of work.

**Comment by GSI** [comment]
> Regarding cryptocurrencies, even though I like the concept from a theoretical perspective, I'm reluctant to participate in any system built on proof of work.

In the context of environmental considerations, I'd like to be so fair as to point out PoW as more sensitive than credit- and debit-card systems, when we consider data centers as well as international office-, support- and communication-infrastructure of both the original providers and their associated payment processors, pollution by employee commutes and the humanitarian aspect of presumably inadequate work conditions in the usual outsourcing-regions.

Within crypto currency, other proof-systems tend to put the power into the hands of those who care the least about human condition.

I can relate 100% to the general desire of wanting to create & give without expecting returns, so:
Thank you for this launcher - it finally makes Android a pleasure to work with :)
